sensor/kubernetes/eventpipeline/resolver/resolver.go

```diff
@@ -0,0 +1,94 @@
+package resolver
+
+import (
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/sensor/kubernetes/eventpipeline/message"
+	"github.com/stackrox/rox/sensor/kubernetes/eventpipeline/output"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+type DeploymentResolver interface {
+	ProcessDependencies(ref message.DeploymentRef) ([]*storage.Deployment, error)
+}
+
+type Resolver interface {
+	Send(pipelineMessage *message.ResourceEvent)
+	SetDeploymentResolver(resolver DeploymentResolver)
+}
+
+type resolverImpl struct {
+	innerQueue                   chan *message.ResourceEvent
+	outputQueue                  output.Queue
+	deploymentDependencyResolver DeploymentResolver
+}
+
+func New(queue output.Queue) Resolver {
+	resolver := &resolverImpl{
+		outputQueue: queue,
+		innerQueue:  make(chan *message.ResourceEvent, 100),
+	}
+	go resolver.startProcessing()
+	return resolver
+}
+
+func (r *resolverImpl) SetDeploymentResolver(resolver DeploymentResolver) {
+	r.deploymentDependencyResolver = resolver
+}
+
+func (r *resolverImpl) startProcessing() {
+	for {
+		// TODO: use a select block + signal to abort
+		msg, more := <-r.innerQueue
+		if !more {
+			return
+		}
+		r.processMessage(msg)
+	}
+}
+
+func (r *resolverImpl) processMessage(pipelineMessage *message.ResourceEvent) {
+	// log.Infof("PROCESSING DEPLOYMENT IDS: %v", pipelineMessage.DeploymentRefs)
+	for _, deploymentRef := range pipelineMessage.DeploymentRefs {
+		// Process deployments
```

Comment:
This is a very good place to add an abstraction for "deployment resolveres":
1. Find deployments based on IDs: used by deployment handlers
2. Find deployments based on service accounts: used by RBAC handlers
3. Find deployments based on labels: used by Network Policies and Services

Reply:
This includes implementing the deployment store functions to do all these lookups.